src/pages/api/clinical-trial-search.ts

```diff
@@ -147,78 +155,81 @@ function buildBundle(searchParams: SearchParameters): Bundle {
     const profileValue = fhirConstants.MCODE_CLINICAL_DISTANT_METASTASIS;
     const codingSystem: string = null;
     const codingSystemCode: string = null;
-    convertStringtoResource({
-      bundle: patientBundle,
-      valueString: stageParm,
+
+    const resource = convertStringToObservation({
+      valueString: metastasisParm,
       id,
       profile_value: profileValue,
       codingSystem,
       codingSystemCode,
     });
+    patientBundle.entry.push({ resource: resource });
   }
 
-  const bioMarkersParm = searchParams.biomarkers;
-  if (bioMarkersParm) {
+  const biomarkers = parseCodedValueArray(searchParams['biomarkers']);
+  if (searchParams.biomarkers.length > 0) {
     const id = 'mcode-tumor-marker';
     const profileValue = fhirConstants.MCODE_TUMOR_MARKER;
-    const codingSystem = 'http://loinc.org';
-    const codingSystemCode = '21907-1';
-    convertStringtoResource({
-      bundle: patientBundle,
-      valueString: metastasisParm,
-      id,
-      profile_value: profileValue,
-      codingSystem,
-      codingSystemCode,
-    });
+    const codingSystem = '';
```

Comment:
Why is this `const` an empty string?

Reply:
replaced empty string with appropriate coding system codes.